lexicon/providers/vultr.py

```diff
@@ -23,85 +24,98 @@ class Provider(BaseProvider):
         }
-        if rtype == "TXT":
-            record["data"] = f'"{content}"'
-        else:
-            record["data"] = content
         if self._get_lexicon_option("ttl"):
             record["ttl"] = self._get_lexicon_option("ttl")
-        self._post("/dns/create_record", record)
 
-        LOGGER.debug("create_record: %s", True)
+        result = self._post(f"/domains/{self.domain_id}/records", record)
+        LOGGER.debug("create_record: %s", result["record"]["id"])
         return True
 
     # List all records. Return an empty list if no records found
     # type, name and content are used to filter records.
     # If possible filter during the query, otherwise filter after response is received.
     def _list_records(self, rtype=None, name=None, content=None):
-        payload = self._get("/dns/records", {"domain": self.domain_id})
+        url = f"/domains/{self.domain_id}/records"
+
+        payload = self._get(url)
+        unprocessed_records = payload["records"]
+
+        while payload["meta"]["links"]["next"] != "":
+            query_params = {"cursor": payload["meta"]["links"]["next"]}
+            payload = self._get(url, query_params=query_params)
+            unprocessed_records.extend(payload["records"])
+
         records = []
-        for record in payload:
+        for record in unprocessed_records:
             processed_record = {
                 "type": record["type"],
-                "name": f"{record['name']}.{self.domain_id}",
-                "ttl": record.get("ttl", self._get_lexicon_option("ttl")),
+                "name": self._full_name(record["name"]),
+                "ttl": record["ttl"],
                 "content": record["data"],
-                "id": record["RECORDID"],
+                "id": record["id"],
             }
             processed_record = self._clean_TXT_record(processed_record)
             records.append(processed_record)
 
         if rtype:
             records = [record for record in records if record["type"] == rtype]
         if name:
-            records = [
-                record for record in records if record["name"] == self._full_name(name)
-            ]
+            records = [record for record in records if record["name"] == self._full_name(name)]
         if content:
             records = [record for record in records if record["content"] == content]
 
         LOGGER.debug("list_records: %s", records)
         return records
 
-    # Create or update a record.
+    # Update a record. Identifier must be specified.
     def _update_record(self, identifier, rtype=None, name=None, content=None):
+        record = {}
 
-        data = {
-            "domain": self.domain_id,
-            "RECORDID": identifier,
-            "ttl": self._get_lexicon_option("ttl"),
-        }
-        # if rtype:
-        #     data['type'] = rtype
+        # type cannot be specified: raise exception if it is set?
+        #if rtype:
+        #    data["type"] = rtype
```

Comment:
As the comment says: the record type cannot be changed using the [vultr update API][1]. So what should be done when the `rtype` variable is set?

[1]: https://www.vultr.com/api/#operation/update-dns-domain-record